src/compiler/emitter.ts

```diff
@@ -5009,7 +5010,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
             
             /** Serializes the return type of function. Used by the __metadata decorator for a method. */
             function emitSerializedReturnTypeOfNode(node: Node): string | string[] {
-                if (node && isFunctionLike(node)) {
+                if (node && isFunctionLike(node) && (<FunctionLikeDeclaration>node).type) {

```

Comment:
I agree with you, I think that the write value of `emitSerializedReturnTypeOfNode` (`void 0`) is the right one when a function has no return type declared; instead, with this fix, `emitSerializedNode` will always write "Object", that could be wrong in some situation. Maybe we could enforce the null check in `emitSerializedNode` (as we did with this fix) **and** each caller should emit the right output by handling the undefined value itself. What do you think about this?